specification/columns/commitmentdiscountquantity.md

```diff
@@ -0,0 +1,64 @@
+# Commitment Discount Quantity
+
+Commitment Discount Quantity is the amount of purchased, used, or unused [*Commitment Discount Units*](#glossary:commitmentdiscountunit) applied to a [*commitment discount's*](#commitment-discount) [*row*](#glossary:row). For a purchase, the Commitment Discount Quantity is either the total or apportioned number of *Commitment Discount Units* over a *commitment discount's* [*term*](#glossary:term). For committed usage, the Commitment Discount Quantity is either the number of *Commitment Discount Units* consumed by a [*row*](glossary:#row) that is covered by a *commitment discount* or is the unused portion of a *commitment discount* over a [*charge period*](#glossary:chargeperiod). The CommitmentDiscountQuantity column only applies to *commitment discounts* and not [*negotiated discounts*](#glossary:negotiated-discount).
```

Comment:
I'm not sure if I've misinterpreted the sample data or if we need to rephrase the following sentence (if yes, I haven't come up with a suggestion yet :( ):

> For a purchase, the Commitment Discount Quantity is either the total of *Commitment Discount Units* over a *commitment discount's* [*term*](#glossary:term) or the apportioned amount for the [*charge period*](#glossary:chargeperiod).

My understanding is that:

 - In the case of a full-upfront payment, it represents the total quantity over the entire term.
 - In the case of a no-upfront payment, it represents the apportioned amount for the charge period.
 - In the case of a partial-upfront payment, it doesn't represent either scenario
 
 For reference, here is a quote from the Commitment Discount Appendix:
  
 > Half of the commitment is billed _once_ during the first charge period of the term for $4,380 (derived as `24 hours * 182.5 days * $1.00`), and the other half is billed across each charge period over the term, derived as (`$1.00 * 8,760 hours * 0.5`)..

Reply:
I think the commitment discount quantity on the purchase is the same as the sum of the apportioned records for the term. So yes, it would be double counted even just on purchases and need to add a note to mention this. The way it made sense to me is you're still purchasing the commitment for the same number of hours - so the quantity is the same for the one-time and sum(periodic records). It's just the rate that's different based on how much was paid upfront vs periodically.

Reply:
For a partial upfront, half the units are on the purchase and the other half are spread over each charge period. The number of eligible units per charge period is 2x what is reported by the recurring purchases.